src/components/SiteCarousel.js

```diff
@@ -0,0 +1,151 @@
+        super(props);
+        this.state = {}
+    }
+
+    componentWillMount() {
+        const {displaySites, selectedSite} = this.props;
+
+        if(displaySites.length === 0){
+            return;
+        }
+
+        const currentSiteIndex = _.findIndex(displaySites, site => site.id === selectedSite.id);
+
+        this.setState({
+            initialIndex: currentSiteIndex > -1 ? currentSiteIndex : 0
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {selectedSite} = this.props;
+
+        if(nextProps.displaySites.length === 0){
+            return;
+        }
+
+        if(selectedSite.id !== nextProps.selectedSite.id){
+            let currentSiteIndex = _.findIndex(nextProps.displaySites, site => site.id === nextProps.selectedSite.id);
+
+            this.setState({
+                initialIndex: currentSiteIndex > -1 ? currentSiteIndex : 0
+            });
+        }
+    }
+
+    selectedSiteIsFavorite = () => {
+        const {currentUser, selectedSite} = this.props;
+
+        return !!_.find(currentUser.favorites, favorite => favorite.id === selectedSite.id);
+    };
+
+    toggleSiteFavorite = ({siteToToggle}) => {
+        const {currentUser} = this.props;
+
+        if (!this.selectedSiteIsFavorite()) {
+            this.props.attemptToAddFavorite({favoriteSiteToAdd: siteToToggle, currentUser});
+        } else {
+            this.props.attemptToRemoveFavorite({favoriteSiteToRemove: siteToToggle, currentUser});
+        }
+    };
+
+    setNextSelectedSite = (newIndex) => {
+        const {displaySites} = this.props;
+
+        const newSelectedSite = displaySites[newIndex];
+
+        this.props.getSiteDetail({selectedSite: newSelectedSite});
+    };
+
+
+    renderSlides = () => {
+        const {navigate, displaySites} = this.props;
+
+        return _.map(displaySites, site => {
+            return (
+                <SitePreview
+                    key={site.key}
+                    navigate={navigate}
+                    siteToPreview={site}
+                    getSiteDetail={this.props.getSiteDetail}
+                    isFavorite={this.selectedSiteIsFavorite()}
+                    toggleSiteFavorite={this.toggleSiteFavorite}
+                />
+            );
+        });
+
+    };
+
+    render() {
+        const {selectedSite} = this.props;
+
+        if (!selectedSite || _.isEmpty(selectedSite)) {
+            return null;
+        }
+
+        const {carouselContainerStyle, swipeButtonRowStyle} = styles;
+        return (
+            <View style={carouselContainerStyle}>
+                <Swiper
+                    showsButtons={true}
+                    buttonWrapperStyle={swipeButtonRowStyle}
+                    index={this.state.initialIndex}
+                    onIndexChanged={this.setNextSelectedSite}
+                    loop={false}
+                >
+                    {this.renderSlides()}
+                </Swiper>
+            </View>
```

Comment:
Nice simple render method where we just call the `Swiper` library, and pass it an array of `SitePreview` components.